snap/squashfs/squashfs.go

```diff
@@ -321,16 +327,21 @@ func (s *Snap) Walk(relative string, walkFn filepath.WalkFunc) error {
 	defer cmd.Process.Kill()
 
 	scanner := bufio.NewScanner(stdout)
-	// skip the header
-	for scanner.Scan() {
-		if len(scanner.Bytes()) == 0 {
-			break
-		}
-	}
-
 	skipper := make(skipper)
+	seenHeader := false
 	for scanner.Scan() {
-		st, err := fromRaw(scanner.Bytes())
+		raw := scanner.Bytes()
+		if !seenHeader {
+			// try to match the header written by older (pre-4.5)
+			// squashfs tools
+			if len(scanner.Bytes()) == 0 ||
+				maybeHeaderRegex.Match(raw) {
+				continue
+			} else {
+				seenHeader = true
```

Comment:
Nitpick, I found seenHeader name a bit misleading in realation to maybeHeaderRegex; it suggests seenHeader becomes true when we match maybeHeaderRegex while afaiu it is set after we scan first lines (which may or may not contain the optional headers). Maybe rename it to something like `pastHeaders` or `headersScanned`?